runtime/compiler/runtime/JITServerAOTCache.hpp

```diff
@@ -338,6 +477,9 @@ class JITServerAOTCache
       const uintptr_t _includedClasses;
       };
 
+   static WellKnownClassesKey getRecordKey(const AOTCacheWellKnownClassesRecord *record)
+      { return { record->records(), record->data().list().length() }; }
```

Comment:
I think this may be the problem. Notice that the key initializer that this replaces above is `{ record->records(), length, includedClasses }`. This is likely the cause of the discrepancies I saw in https://github.com/eclipse-openj9/openj9/pull/15949#issuecomment-1282701588, and might be the cause of the segfault as well.

Reply:
Yeah that's probably it. This could have been prevented by defining actual constructors in the `...Key` structs, which is still worth doing.